### PR TITLE
Add alternative make_line function with an example

### LINKED ISSUE #4174: [FEATURE] Add make_line expression function variant which accepts
an array of points
Original commit: https://github.com/qgis/QGIS/commit/80d77b4788a08b8daa9782e3b99de4c56b48c074 by nyalldawson

Allows creation of lines from variable numbers of points, and
from sequences from aggregates/dynamically generated sequences

Fixes #31268

### PR DESCRIPTION
**Pull request by DelazJ**
Fix #4174

### CODE REVIEW

File: source/docs/user_manual/working_with_vector/expression.rst [hunk 1]
```diff
@@ -759,6 +754,24 @@ This group contains functions that operate on geometry objects (e.g., length, ar
...
+
+    buffer( $geometry, 10 )
+    point_on_surface( $geometry )
+
+* Given a point feature, generate a closed line to whom it's the center point::
+
+    make_line(
+      -- using an array of points placed around the original
+      array_foreach(
+        -- list of angles to place a projected point (every 90°)
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/expression.rst]:
```suggestion
        -- list of angles for placing the projected points (every 90°)
```

File: source/docs/user_manual/working_with_vector/expression.rst [hunk 2]
```diff
@@ -759,6 +754,24 @@ This group contains functions that operate on geometry objects (e.g., length, ar
...
+        array:=generate_series( 0, 360, 90 ),
+        -- translate point for 20 units using the associated angle
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/expression.rst]:
```suggestion
        -- translate the point 20 units in the given direction (angle)
```

File: source/docs/user_manual/working_with_vector/expression.rst [hunk 3]
```diff
@@ -759,6 +754,24 @@ This group contains functions that operate on geometry objects (e.g., length, ar
 
     CASE WHEN $area > 10 000 THEN 'Larger' ELSE 'Smaller' END
 
+* You can manipulate the current geometry with the variable $geometry to create
+  a buffer or get the point on surface::
```
[the remark this answers is not in the document]
Comment (DelazJ) [reply source/docs/user_manual/working_with_vector/expression.rst]: I thought "point on surface" was a particular point and not simply a point within the geometry

[the remark this answers is not in the document]
Comment (havatv) [reply source/docs/user_manual/working_with_vector/expression.rst]: The position of the "point on surface" depends on the algorithm used. In this way it is different from the centroid, which is well defined.

**Pull request merged**